src/cloudant/feed.py

```diff
@@ -206,3 +206,37 @@ def _process_data(self, line):
         except ValueError:
             data = {"error": "Bad JSON line", "line": line}
         return skip, data
+
+class InfiniteFeed(Feed):
+    """
+    Provides an infinite iterator for consuming client and database feeds such
+    as ``_db_updates`` and ``_changes``.  An InfiniteFeed object is constructed
+    with a reference to a client or database Requests Session object and a feed
+    endpoint URL.  Unlike a :class:`~cloudant.feed.Feed` which can be either a
+    ``normal``, ``longpoll``, or ``continuous`` feed an InfiniteFeed can only be
+    ``continuous`` and the iterator will only stream formatted JSON objects.
+    Instead of using this class directly, it is recommended to use the client
+    API :func:`~cloudant.client.CouchDB.db_updates` and the database API
+    :func:`~cloudant.database.CouchDatabase.changes`.  Reference those methods
+    for a valid list of feed options.
+
+    :param Session session: Requests session used by the Feed.
+    :param str url: URL used by the Feed.
+    """
+    def __init__(self, session, url, **options):
+        super(InfiniteFeed, self).__init__(session, url, False, **options)
+        # feed must be set to "continuous" in the case of an infinite feed
+        self._options.update({'feed': 'continuous'})
+
+    def next(self):
+        """
+        Handles the iteration by pulling the next line out of the stream and
+        converting the response to JSON.
+
+        :returns: Data representing what was seen in the feed
+        """
+        if self._last_seq:
+            self._options.update({'since': self._last_seq})
+            self._resp = None
+            self._last_seq = None
+        return super(InfiniteFeed, self).next()
```

Comment:
I wonder if this should be wrapped in some exception swallowing (possibly outputting a warning or something). It seems to me that the likely use case of an infinite feed is to keep processing changes _ad infinitum_. Won't it terminate at the first exception atm? Admittedly different exceptions might warrant different behaviour (e.g. a networking glitch probably shouldn't break the feed, but an unauthorized probably should). With that in mind maybe it is sufficient for a user's code to restart the feed after an exception if we explicitly document that it is not exception resilient (and possibly how to restart from the same place)?